pkg/storage/chunk/cache/groupcache.go

```diff
@@ -86,11 +75,13 @@ func NewGroupCache(rm ringManager, config GroupCacheConfig, server *server.Serve
 	addr := fmt.Sprintf("http://%s", rm.Addr())
 	level.Info(logger).Log("msg", "groupcache local address set to", "addr", addr)
 
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	transport.MaxIdleConnsPerHost = 100
```

Comment:
Not sure if we also need to set any timeouts? I can't remember if the default transport has reasonable timeouts

Reply:
Defaults are per se looking like this:
```go
var DefaultTransport [RoundTripper](https://pkg.go.dev/net/http#RoundTripper) = &[Transport](https://pkg.go.dev/net/http#Transport){
	Proxy: [ProxyFromEnvironment](https://pkg.go.dev/net/http#ProxyFromEnvironment),
	DialContext: defaultTransportDialContext(&[net](https://pkg.go.dev/net).[Dialer](https://pkg.go.dev/net#Dialer){
		Timeout:   30 * [time](https://pkg.go.dev/time).[Second](https://pkg.go.dev/time#Second),
		KeepAlive: 30 * [time](https://pkg.go.dev/time).[Second](https://pkg.go.dev/time#Second),
	}),
	ForceAttemptHTTP2:     [true](https://pkg.go.dev/builtin#true),
	MaxIdleConns:          100,
	IdleConnTimeout:       90 * [time](https://pkg.go.dev/time).[Second](https://pkg.go.dev/time#Second),
	TLSHandshakeTimeout:   10 * [time](https://pkg.go.dev/time).[Second](https://pkg.go.dev/time#Second),
	ExpectContinueTimeout: 1 * [time](https://pkg.go.dev/time).[Second](https://pkg.go.dev/time#Second),
}
```